src/node_crypto.cc

```diff
@@ -2769,7 +2772,11 @@ bool CipherBase::InitAuthenticated(const char* cipher_type, int iv_len,
   }
 
   const int mode = EVP_CIPHER_CTX_mode(ctx_.get());
-  if (mode == EVP_CIPH_CCM_MODE || mode == EVP_CIPH_OCB_MODE) {
+  bool needs_auth_tag_length = mode == EVP_CIPH_CCM_MODE;
+#ifndef OPENSSL_NO_OCB
+  needs_auth_tag_length |= mode == EVP_CIPH_OCB_MODE;
```

Comment:
Some people won't be happy about using this operator I guess.

Reply:
I guess the alternative would be something like

```c++
bool needs_auth_tag_length = mode == EVP_CIPH_CCM_MODE
#ifndef OPENSSL_NO_OCB
    || mode == EVP_CIPH_OCB_MODE;
#endif
  ;
```

? Not that much prettier ;)

Reply:
I am fine with the change as it is, but other collaborators have expressed concerns about mixing int/bool operators in the past :)

Reply:
I don't care about the operator mixing, but I very mildly prefer @addaleax's idiom above, and would use something similar to it for the block around line 2547 as well: a list of `|| blah` statements, some of them with ifdef's around them.